src/libsystemd/sd-device/sd-device.c

```diff
@@ -1943,29 +1943,14 @@ _public_ int sd_device_set_sysattr_value(sd_device *device, const char *sysattr,
                 return -ENOMEM;
 
         r = write_string_file(path, value, WRITE_STRING_FILE_DISABLE_BUFFER | WRITE_STRING_FILE_NOFOLLOW);
-        if (r < 0) {
-                if (r == -ELOOP)
-                        return -EINVAL;
-                if (r == -EISDIR)
-                        return r;
-
-                r = free_and_strdup(&value, "");
-                if (r < 0)
-                        return r;
-
-                r = device_add_sysattr_value(device, sysattr, value);
```

Comment:
Hmm, equivalent code was already there when this function was initially added in 57fa1d094cd2c5ac68970526ad0a0754c548e75d. Yeah, it seems better to drop this strange thing.